### PR TITLE
Add a conversions header

### PR DESCRIPTION
**Pull request by haudren**
This is a follow-up to #14 , and replaces it. It adds support for homogeneous matrices, with a test. Maybe we should also add support for Eigen Transforms ? I guess this can come in a later PR.

### CODE REVIEW

**Comment by haudren** [comment]
@gergondet : Is this conform to your expectations? I think it still needs a bit of documentation, but the functionality is here.

File: src/SpaceVecAlg/Conversions.h [hunk 1]
```diff
@@ -0,0 +1,78 @@
+// Copyright 2012-2016 CNRS-UM LIRMM, CNRS-AIST JRL
```
Comment (gergondet) [inline src/SpaceVecAlg/Conversions.h]: 2017 :calendar:

File: src/SpaceVecAlg/Conversions.h [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      return PTransform<T>(m.template block<3,3>(0,0).transpose(), m.template block<3,1>(0,3));
+    }
+    else
+    {
+      return PTransform<T>(m.template block<3,3>(0,0), m.template block<3,1>(0,3));
+    }
+  }
+
+  template<typename T>
+  PTransform<T> fromHomogeneous(const Eigen::Matrix<T, 4, 4>& m,
```
Comment (gergondet) [inline src/SpaceVecAlg/Conversions.h]: Doesn't this fall under the previous function?

File: src/SpaceVecAlg/Conversions.h [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    {
+      return PTransform<T>(m.template block<3,3>(0,0), m.template block<3,1>(0,3));
+    }
+  }
+
+  template<typename T>
+  Eigen::Matrix<T, 4, 4> toHomogeneous(const PTransform<T>& pt,
+                                       bool rightHandedness = RightHanded)
+  {
+    Eigen::Matrix<T, 4, 4> res = Eigen::Matrix<T, 4, 4>::Identity();
```
Comment (gergondet) [inline src/SpaceVecAlg/Conversions.h]: Maybe
```cpp
Eigen::Matrix<T, 4, 4> res = Eigen::Matrix<T, 4, 4>::Zero();
res(3, 3) = 1.0;
```
is slightly more efficient.

File: tests/ConversionsTest.cpp [hunk 4]
```diff
@@ -0,0 +1,60 @@
+// Copyright 2012-2016 CNRS-UM LIRMM, CNRS-AIST JRL
```
Comment (gergondet) [inline tests/ConversionsTest.cpp]: 2017 :calendar:

File: tests/ConversionsTest.cpp [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+#include <boost/math/constants/constants.hpp>
+
+// SpaceVecAlg
+#include <SpaceVecAlg/SpaceVecAlg>
+#include <SpaceVecAlg/Conversions.h>
+
+BOOST_AUTO_TEST_CASE(ConverionsHomogeneous)
+{
+	using namespace Eigen;
+	using namespace sva;
```
Comment (gergondet) [inline tests/ConversionsTest.cpp]: Wrong indentation (in the line above as well)

File: tests/ConversionsTest.cpp [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+#define BOOST_TEST_DYN_LINK
+#define BOOST_TEST_MODULE PTransformd test
+#include <boost/test/unit_test.hpp>
+#include <boost/math/constants/constants.hpp>
+
+// SpaceVecAlg
+#include <SpaceVecAlg/SpaceVecAlg>
+#include <SpaceVecAlg/Conversions.h>
+
+BOOST_AUTO_TEST_CASE(ConverionsHomogeneous)
```
Comment (gergondet) [inline tests/ConversionsTest.cpp]: Typo in the test case name.

File: tests/ConversionsTest.cpp [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+{
+	using namespace Eigen;
+	using namespace sva;
+
+    Matrix4d hom = Matrix4d::Zero();
+    hom(0,1) = -1;
+    hom(1,0) = +1;
+    hom(2,2) = +1;
+    hom(3,3) = +1;
+    PTransform<double> pt = conversions::fromHomogeneous(hom,conversions::RightHanded);
```
Comment (gergondet) [inline tests/ConversionsTest.cpp]: Could be `PTransformd` to save a few chars.

Applies to all instances below.

**Review by gergondet — CHANGES_REQUESTED** [review]

File: src/SpaceVecAlg/Conversions.h (hunk 2 above)
Comment (haudren) [reply src/SpaceVecAlg/Conversions.h]: I would have said so, I had not noticed that @barcode had a second implementation.

Comment (haudren) [reply src/SpaceVecAlg/Conversions.h]: Actually, I just checked the code fails to compile here if the second definition is not present:
```
‘Eigen::Matrix4d {aka Eigen::Matrix<double, 4, 4>}’ is not derived from ‘const Eigen::Ref<Eigen::Matrix<T, 4, 4> >’
```

**Comment by haudren** [comment]
Ok I fixed the aforementioned points, just need to add a little documentation and we should be good to go.

**Review by gergondet — APPROVED** [review]

**Pull request merged**